csrc/options.h

```diff
@@ -0,0 +1,83 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#pragma once
+
+#include <c10/macros/Export.h>
+
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+namespace nvfuser {
+
+//! Types of features to enable
+//!
+//! These can be set through the `PYTORCH_NVFUSER_ENABLE` environment variable
+//!
+enum class EnableOption {
+  Complex, //! Enable complex support on python
+  ConvDecomposition, //! Enable conv-bias decomposition
+  GraphOp, //! Enable graphOps(index_select/gather/scatter)
+  KernelDb, //! Enable Kernel Database
+  KernelProfile, //! Enable intra-kernel performance profiling
+  LinearDecomposition, //! Enable linear-bias decomposition
+  WarnRegisterSpill, //! Enable warnings of register spill
+  EndOfOption //! Placeholder for counting the number of elements
+};
+
+//! The base template class for the options such as EnableOption
+template <typename OptionEnum>
+class Options {
+ public:
+  Options() = default;
+
+  bool has(OptionEnum option) const {
+    return options_.count(option);
+  }
+
+  const std::vector<std::string>& getArgs(OptionEnum option) const {
+    TORCH_INTERNAL_ASSERT(has(option), "Option not set");
+    return options_.at(option);
+  }
+
+  void set(OptionEnum option_type, std::vector<std::string> option = {}) {
+    options_[option_type] = option;
+  }
+
+  void unset(OptionEnum option_type) {
+    options_.erase(option_type);
+  }
+
+ protected:
+  std::unordered_map<OptionEnum, std::vector<std::string>> options_;
+};
+
+class TORCH_CUDA_CU_API EnableOptions : public Options<EnableOption> {
+ public:
+  EnableOptions();
+};
+
+//! Utility class to temporarily overrride the Enable options
```

Comment:
It would be good to mention that this guard has higher priority than the env:
```suggestion
//! Utility class to temporarily overrride the Enable options provided by the env
```